packages/terra-form-select/src/single/Frame.jsx

```diff
@@ -396,12 +382,29 @@ class Frame extends React.Component {
     return ariaLabel === undefined ? defaultAriaLabel : ariaLabel;
   }
 
+  /**
+   * Generates the main label for the field.
+   */
+  mainLabel() {
+    const selectedText = this.props.intl.formatMessage({ id: 'Terra.form.select.selected' });
+    let label;
+    if (this.props.display) {
+      label = `${this.props.display} ${selectedText}, ${this.ariaLabel()}`;
+    } else if (this.props.placeholder) {
+      label = `${this.props.placeholder}, ${this.ariaLabel()}`;
+    }
+
+    return label || this.ariaLabel();
+  }
+
   /**
    * Determines compatible role attribute to apply to select based on active variant and disabled prop
    */
   role() {
     const { disabled } = this.props;
-    return disabled ? undefined : 'combobox';
+    const role = SharedUtil.isSafari() ? 'group' : 'button';
```

Comment:
```suggestion
    const role = SharedUtil.isSafari() ? 'group' : 'combobox';
```
Should this be combobox instead of button ..?? 
Because this would make VO to read form-select as button when it focused with keyboard.

Reply:
This was set to "button" for chrome as VO had issues wherein it was announcing the selected item twice. This is because `aria-hidden` wasn't taking effect when role was set to "combobox". With this change, though the role is button, the custom response would mention "Expanded/Collapsed combobox" first and at the end of response it would finally read as button. However this issue doesn't occur in safari, as it's using `aria-live=polite`

Reply:
Yes, tested on edge.